### PR TITLE
fix sort order in distributed RETURN DISTINCT

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: arangod/Aql/OptimizerRules.cpp [hunk 1]
```diff
@@ -140,6 +140,59 @@ arangodb::aql::Variable const* getOutVariable(arangodb::aql::ExecutionNode const
...
+          continue;
+        }
+        other.clear();
+        try {
+          buffer.clear();
+          expr->stringify(&buffer);
+          other = std::string(buffer.c_str(), buffer.size());
+        } catch (...) {
+        }
+        if (other == cmp) {
```
Comment (graetzer) [inline arangod/Aql/OptimizerRules.cpp]: can't you compare with the cstr ?

**Comment by graetzer** [comment]
are the tests here already in devel: https://github.com/arangodb/arangodb/pull/8934#pullrequestreview-235602671 ?

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/4408/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/4417/

**Comment by jsteemann** [comment]
@graetzer : yes, the tests are in devel already. They have been added by a different PR before.

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/4456/

**Pull request merged**